### PR TITLE
Development

### PR DESCRIPTION
**Pull request by kelvinwijaya**
# Description

Bugs fix for following items:
- During Base String generation, handle empty value of the key value pair by appending & value
- Standardise nonce value generation to use 32 bytes and base64 encoded string
- Verify the support of multiple param with same key

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code enhancement and update (non-breaking change)

## How Has This Been Tested?

Tested using unit test-cases

### CODE REVIEW

File: src/main/java/com/api/util/ApiSecurity/ApiSigning.java [hunk 1]
```diff
@@ -499,7 +501,7 @@ public static String getSignatureToken(
 
             // Generate the nonce value
             try {
-                nonce = nonce != null ? nonce : Long.toString(getNewNonce());
+                nonce = nonce != null ? nonce : getNewNonce();
```
Comment (robincher) [inline src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: Maybe check if nonce is empty string also

File: src/main/java/com/api/util/ApiSecurity/ApiSigning.java [hunk 2]
```diff
@@ -411,16 +408,21 @@ public static String getBaseString(String authPrefix
...
 
                 String[] paramArr = queryString.split("&");
                 for (String item : paramArr) {
-                    log.debug("item:: {}", item);
+                    log.debug("queryItem:: {}", item);
                     String[] itemArr = item.split("=");
                     try {
-                        paramList.add(itemArr[0], java.net.URLDecoder.decode(itemArr[1], StandardCharsets.UTF_8.toString()));
+                    	if(itemArr.length == 1) {
+                    		paramList.add(itemArr[0], "");
```
Comment (robincher) [inline src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: It will be passing as **key = ''** as an empty string for the basestring signing?

Comment (kelvinwijaya) [reply src/main/java/com/api/util/ApiSecurity/ApiSigning.java]: The BaseString will be resolve to: http://localhost/api/resource?key&key2=123

So , equal sign will be dropped

File: src/main/java/com/api/util/ApiSecurity/ApiList.java [hunk 3]
```diff
@@ -43,7 +43,7 @@ public String toString(String delimiter, Boolean sort, Boolean quote)
 						return l1.getKey().equals(l2.getKey()) ? l1.getValue().compareTo(l2.getValue())
 									: l1.getKey().compareTo(l2.getKey());
 					})
-					.map(e -> String.format(format, e.getKey(), e.getValue()))
+					.map(e -> (e.getValue()=="" && isBaseString) ? e.getKey() : String.format(format, e.getKey(), e.getValue())  )
```
[the remark this answers is not in the document]
Comment (kelvinwijaya) [reply src/main/java/com/api/util/ApiSecurity/ApiList.java]: will fix this

**Pull request merged**